src/frontend/src/scheduler/hummock_snapshot_manager.rs

```diff
@@ -61,9 +62,26 @@ enum EpochOperation {
     Tick,
 }
 
+pub struct HummockSnapshotGuard {
+    pub snapshot: HummockSnapshot,
+    query_id: QueryId,
+    unpin_snapshot_sender: UnboundedSender<EpochOperation>,
+}
+
+impl Drop for HummockSnapshotGuard {
+    fn drop(&mut self) {
+        self.unpin_snapshot_sender
+            .send(EpochOperation::ReleaseEpoch {
+                query_id: self.query_id.clone(),
+                epoch: self.snapshot.committed_epoch,
+            })
+            .expect("Unpin channel should never closed");
+    }
+}
+
 impl HummockSnapshotManager {
     pub fn new(meta_client: Arc<dyn FrontendMetaClient>) -> Self {
-        let (sender, mut receiver) = channel(MAX_WAIT_EPOCH_REQUEST_NUM);
+        let (sender, mut receiver) = tokio::sync::mpsc::unbounded_channel();
```

Comment:
Why remove the `MAX_WAIT_EPOCH_REQUEST_NUM` on the channel?

Reply:
Because `drop` is not async, but bounded channel's `send` is async. So I have to switch to `UnboundedChannel` as for `PinnedVersion`.